lib/make-badge-url.js

```diff
@@ -0,0 +1,30 @@
+'use strict'
+
+const queryString = require('query-string')
+
+function encodeField(s) {
+  return encodeURIComponent(s.replace(/-/g, '--').replace(/_/g, '__'))
+}
+
+function staticBadgeUrl({
+  baseUrl,
+  label,
+  message,
+  color = 'lightgray',
+  style,
+}) {
+  if (!label || !message) {
+    throw Error('label and message are required')
+  }
+  const path = [label, message, color].map(encodeField).join('-')
+  const outQueryString = queryString.stringify({
+    style,
+  })
+  const suffix = outQueryString ? `?${outQueryString}` : ''
+  return `/badge/${path}.svg${suffix}`
```

Comment:
We could make this slightly more general by accepting a format param (which also accepts `.gif`, `.png` etc) with a default of `.svg` if you wanted.

Reply:
Sounds good.